fhirflat/flat2fhir.py

```diff
@@ -9,13 +13,33 @@
 from fhir.resources.backbonetype import BackboneType as _BackboneType
 
 from pydantic.v1.error_wrappers import ValidationError
+from pydantic.v1 import BaseModel
 
 
 def create_codeable_concept(
     old_dict: dict[str, list[str] | str], name: str
 ) -> dict[str, list[str]]:
     """Re-creates a codeableConcept structure from the FHIRflat representation."""
-    codes = old_dict.get(name + ".code")
+
+    # for reading in from ingestion pipeline
+    if name + ".code" in old_dict and name + ".system" in old_dict:
+        raw_codes = old_dict.get(name + ".code")
+        if not isinstance(raw_codes, list):
+            formatted_code = (
+                raw_codes if isinstance(raw_codes, str) else str(int(raw_codes))
+            )
+            codes = [old_dict[name + ".system"] + "|" + formatted_code]
+        else:
+            formatted_code = [
+                c if isinstance(c, str) else str(int(c)) for c in raw_codes
```

Comment:
This assumes that leading 0s in codes are not significant - is that true? (probably, but good to make sure)

Reply:
Oh, not sure about this - I don't think I've seen codes with leading 0's but that doesn't mean they don't exist

Reply:
@sadiekelly have you seen any codes starting with leading 0s?

Reply:
No, I haven't, none of the codes I've used so far begin with a zero. I've not been able to find anything that definitively says codes do not start with a zero, but it seems like a fair assumption to make from what we currently are using. I'll keep an eye out for any codes that do have leading zeros.

Reply:
Ok, will create a ticket for this